roles/certificates/tasks/main.yml

```diff
@@ -0,0 +1,54 @@
+---
+- name: deploy CA public and private keys
+  sudo: yes
+  copy:
+    src: "{{ item.src }}"
+    dest: "{{ item.dest }}"
+    owner: root
+  with_items:
+    # CA public key
+    - src: ssl/cacert.pem
+      dest: /etc/pki/CA/ca.cert
+    - src: ssl/cacert.pem # https://github.com/asteris-llc/mantl-packaging/issues/90
+      dest: /etc/pki/CA/cacert.pem
+    - src: ssl/cacert.pem
+      dest: /etc/pki/ca-trust/source/anchors/cacert.pem
+    # CA private key
+    - src: ssl/private/cakey.pem
+      dest: /etc/pki/CA/ca.key
+    - src: ssl/private/cakey.pem
+      dest: /etc/pki/CA/private/cakey.pem
+  notify:
+    - update-ca-trust
+
+- name: install generate-certificate
+  sudo: yes
+  yum:
+    name: generate-certificate-0.0.1
+    state: present
+
+- name: create certificate destination directory
+  sudo: yes
+  file:
+    state: directory
+    path: "{{ destination }}"
+    owner: root
+    group: root
+    mode: 755
+
+- name: generate certificates
+  sudo: yes
+  command: >
+    generate-certificate
+    {% if country != "" %} --country "{{ country }}" {% endif %}
+    {% if state != "" %} --state "{{ state }}" {% endif %}
+    {% if locality != "" %} --locality "{{ locality }}" {% endif %}
+    {% if organization != "" %} --organization "{{ organization }}" {% endif %}
+    {% if unit != "" %} --unit "{{ unit }}" {% endif %}
+    {% if email != "" %} --email "{{ email }}" {% endif %}
+    {% if dns != "" %} --dns "{{ dns | unique | join(' ') }}" {% endif %}
+    {% if ip != "" %} --ip "{{ ip | unique | join(' ') }}" {% endif %}
+    --location "{{ location }}"
+    --destination "{{ destination }}"
+  args:
+    creates: "{{ destination }}/{{ prefix }}{% if prefix != '' %}.{% endif %}key"
```

Comment:
The role could be reused in e.g. an addon, where you might want to change `prefix`.